### PR TITLE
Update dependency

### PR DESCRIPTION
**Pull request by baluchicken**
| Q               | A
| --------------- | ---
| Bug fix?        | no|
| New feature?    |yes|
| API breaks?     | no|
| Deprecations?   | no|
| License         | Apache 2.0

### What's in this PR?

- This PR updates the operators dependencies to the latest available.
- Moves the API to a different module to make easier to use it as a lib.
- Fixes controller gen error, by moving it to the bin directory instead of using the system wide one.
- Fixes watcher error which kills the controller runtime if the resource is not available on the cluster

### Checklist

- [x] Implementation tested
- [x] Error handling code meets the [guideline](https://github.com/banzaicloud/pipeline/blob/master/docs/error-handling-guide.md)
- [x] Logging code meets the guideline
- [x] User guide and development docs updated (if needed)

### CODE REVIEW

File: config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml [hunk 1]
```diff
@@ -997,12 +529,8 @@ spec:
                 cruiseControlEndpoint:
                   type: string
                 cruiseControlTaskSpec:
-                  description: CruiseControlTaskSpec specifies the configuration of
-                    the CC Tasks
                   properties:
                     RetryDurationMinutes:
-                      description: RetryDurationMinutes describes the amount of time
```
Comment (amuraru) [inline config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml:1004]: is the removal of description fields on-purpose or a side effect of version upgrade?

Comment (baluchicken) [reply config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml:1004]: Good catch, I will check this this is a side effect of upgrading to the 0.2.5 generator tool. Will check.

Comment (baluchicken) [reply config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml:1004]: I found the problem will fix it on the next commit.

Comment (amuraru) [reply config/base/crds/kafka.banzaicloud.io_kafkaclusters.yaml:1004]: awesome - thanks Balint!

**Pull request merged**